python/GafferUI/MetadataWidget.py

```diff
@@ -347,3 +347,32 @@ def __buttonClicked( self, widget ) :
 		if chosenPath is not None :
 			self.__path.setFromString( str( chosenPath ) )
 			self.__editingFinished()
+
+class NumericMetadataWidget( MetadataWidget ) :
+
+	def __init__( self, key, target = None, defaultValue = 0, **kw ) :
+
+		self.__numericWidget = GafferUI.NumericWidget( value = defaultValue )
+
+		self.__defaultValue = defaultValue
+
+		MetadataWidget.__init__( self, self.__numericWidget, key, target, defaultValue = defaultValue, **kw )
+
+		self.__numericWidget.editingFinishedSignal().connect(
+			Gaffer.WeakMethod( self.__editingFinished ), scoped = False
+		)
+		self.__numericWidget.valueChangedSignal().connect(
+			Gaffer.WeakMethod( self.__editingFinished ), scoped = False
+		)
```

Comment:
There's a bit of tension between the immediate updates we get from `valueChangedSignal()` vs the many many undo steps we end up making for the virtual sliders and cursor-key increments. Might be worth spending half an hour or so to see if we can merge those undos into one, along the lines of what we do in NumericPlugValueWidget?

Reply:
I took a stab at the NumericPlugValueWidget style undo merging in 69d4316bda986880b57b8710c4ad70658f87e40a. 

Should a reason of `ValueChangedReason.InvalidEdit` result in an update using the current metadata value? Currently it is using the default value for the widget for lack of a better value.

Reply:
I think using the default value is reasonable enough given how much of a corner case this is. We're not guaranteed to have a current value, but I suppose if we do have one it might be considered more appropriate to revert to that than the default? I don't mind too much...